src/core/transpile.jl

```diff
@@ -1844,6 +1844,63 @@ end
 
 struct ReadoutsAreFinalInstructionsTranspiler <: Transpiler end
 
+"""
+    transpile(::ReadoutsAreFinalInstructionsTranspiler, circuit::QuantumCircuit)::QuantumCircuit
+
+Ensures that each `Readout` `Instruction` is the last operation 
+on each qubit where readouts are present, and that repeated readouts 
+on the same qubit do not occur, or throws an error. 
+It leaves the `QuantumCircuit` unchanged.
+
+# Examples
+```jldoctest
+julia> transpiler = ReadoutsAreFinalInstructionsTranspiler();
+
+julia> circuit = QuantumCircuit(qubit_count=2, instructions=[hadamard(1), readout(1,1)])
+Quantum Circuit Object:
+   qubit_count: 2 
+   bit_count: 2 
+q[1]:──H────✲──
+               
+q[2]:──────────
+               
+
+julia> transpiled_circuit = transpile(transpiler, circuit)
+Quantum Circuit Object:
+   qubit_count: 2 
+   bit_count: 2 
+q[1]:──H────✲──
+               
+q[2]:──────────
+               
+
+julia> circuit = QuantumCircuit(qubit_count=2, instructions=[hadamard(1), readout(1,1), sigma_x(1)])
+Quantum Circuit Object:
+   qubit_count: 2 
+   bit_count: 2 
+q[1]:──H────✲────X──
+                    
+q[2]:───────────────
+                    
+
+julia> transpiled_circuit = transpile(transpiler, circuit)
+ERROR: AssertionError: Cannot preform Gate following Readout on qubit: 1
+[...]
+
+julia> circuit = QuantumCircuit(qubit_count=2, instructions=[readout(1,1), readout(1,2)])
+Quantum Circuit Object:
+   qubit_count: 2 
+   bit_count: 2 
+q[1]:──✲────✲──
+               
+q[2]:──────────
+               
+
+julia> transpiled_circuit = transpile(transpiler, circuit)
+ERROR: AssertionError: Found multiple Readout on qubit: 1
```

Comment:
```suggestion
ERROR: AssertionError: Found multiple `Readouts` on qubit: 1
```

Reply:
updated, along with other places we printout errors in the docs